Sources/Plasma/PubUtilLib/plSurface/hsGMaterial.cpp

```diff
@@ -202,6 +207,13 @@ void hsGMaterial::SetLayer(plLayerInterface* layer, int32_t which, bool insert,
     }
 }
 
+#if PLASMA_PIPELINE_GL || PLASMA_PIPELINE_METAL
+void hsGMaterial::SetDeviceRef(hsGDeviceRef* ref)
+{
+    hsRefCnt_SafeAssign(fDeviceRef, ref);
+}
```

Comment:
This could be in the header.

Reply:
Finally came back to this one - hsRefCnt_SafeAssign is a macro that would be unpacked in the header - but hsGDeviceRef is forward declared in the header. So this causes an incomplete type error. I could move the include into the header - but I'm not sure we want to increase that headers propagation.

Reply:
`hsGDeviceRef.h` is a small pnNucleusInc header, so it should be fine.